vault/login_mfa.go

```diff
@@ -1114,6 +1119,51 @@ func (b *LoginMFABackend) mfaMethodList(ctx context.Context, methodType string)
 	return keys, configInfo, nil
 }
 
+func (b *LoginMFABackend) mfaLoginEnforcementList(ctx context.Context) ([]string, map[string]interface{}, error) {
+	ns, err := namespace.FromContext(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ws := memdb.NewWatchSet()
+	txn := b.db.Txn(false)
+
+	// get all the login enforcements in our namespace
+	iter, err := txn.Get(memDBMFALoginEnforcementsTable, "namespace", ns.ID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to fetch iterator for login enforcement configs in memdb: %w", err)
+	}
+
+	ws.Add(iter.WatchCh())
+
+	var keys []string
+	enforcementInfo := map[string]interface{}{}
+
+	for {
```

Comment:
Should we care about namespaces? Since we can read enforcement configs only in the same namespace, would this become a work around if we don't check some sort of namespace matching?

Reply:
I'm sorry, I'm not following you. Can you elaborate? Is your comment about this `for` loop specifically or just about this file in general?

Reply:
Sorry about that. Looking at other list functions we have, I noticed that namespace checking is done in this similar for loop, so I decided to add a comment in where it is now. Anyways, the comment is all about our strategy to check for namespaces.

Reply:
The initial `Get` call above on line 1132 should retrieve all of the records where the namespace ID matches the namespace we're in. Nick and I chatted in slack and decided that since login enforcement names were only unique per namespace, it would be confusing to show login enforcement from parent or child namespaces, so only login enforcements from the current namespace should be listed. I believe line 1132 should cover that.

But, I can manually apply this patch to the ENT repo and write a test there to verify this is the case.

Reply:
Ah, I see. That is great! Thanks for the explanation Josh!